JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -481,15 +488,16 @@ - (UICollectionViewCell *)collectionView:(JSQMessagesCollectionView *)collection
     cell.delegate = collectionView;
 
     if (!isMediaMessage) {
-        cell.textView.text = [messageItem text];
-
-        if ([UIDevice jsq_isCurrentDeviceBeforeiOS8]) {
-            //  workaround for iOS 7 textView data detectors bug
+        if ([messageItem conformsToProtocol:@protocol(JSQMessageAttributedData)])
+        {
+            id <JSQMessageAttributedData> attributedMessageItem =  (id <JSQMessageAttributedData>) messageItem;
+            cell.textView.attributedText = [attributedMessageItem attributedText];
+        }
+        else
+        {
             cell.textView.text = nil;
```

Comment:
I am not sure why but it will change the message text and remove all the text after any special character. to avoid that I had to convert all of them to attributed text. As you can see in this picture it removes all the text for some messages
![screen shot 2015-12-07 at 10 34 19 am](https://cloud.githubusercontent.com/assets/9285802/11631018/2108b128-9cce-11e5-8981-6c79a194fcff.png)